<launch>

    <arg name="husky_name" default="husky" doc="Name of the Husky robot for topic namespacing"/>
    <arg name="waypoint_filename" default="optitrack_waypoints_06.txt" doc="Name of the waypoint file in husky_waypoint_nav/config/waypoints/"/>
    <arg name="odom_topic_name" default="/odom" doc="Topic name for the robot's odometry data (nav_msgs/Odometry)"/>
    <arg name="mocap_topic_name" default="/natnet_ros/$(arg husky_name)/pose" doc="Topic name for the MoCap pose data (geometry_msgs/PoseStamped)"/>
    <arg name="cmd_vel_topic_name" default="/husky_velocity_controller/cmd_vel" doc="Topic name for publishing velocity commands"/>
    <arg name="log_dir" default="$(find husky_waypoint_nav)/logs" doc="Directory to save the navigation log files"/>

    <node name="p_navigator_node" pkg="husky_waypoint_nav" type="p_navigator.py" output="screen">

        <param name="husky_name" value="$(arg husky_name)" />
        <param name="mocap_pose_topic" value="$(arg mocap_topic_name)" />
        <param name="odom_topic" value="$(arg odom_topic_name)" />
        <param name="cmd_vel_topic" value="$(arg cmd_vel_topic_name)" />

        <param name="waypoint_file" value="$(arg waypoint_filename)" />
        <param name="log_file_path" value="$(arg log_dir)" />

        <param name="linear_kp" value="0.5" />  
        <param name="angular_kp" value="1.5" /> 
        <param name="max_linear_vel" value="0.5" />  
        <param name="max_angular_vel" value="1.0" /> 
        <param name="pos_tolerance" value="0.1" />   
        <param name="angle_tolerance" value="0.1" /> 
        <param name="rate" value="60" /> 
        </node>

</launch>